src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -645,7 +645,12 @@ public SessionResultsBundle getSessionResultsForUser(
         if (questionId == null) {
             allQuestions = fqLogic.getFeedbackQuestionsForSession(feedbackSessionName, courseId);
         } else {
-            allQuestions = Collections.singletonList(fqLogic.getFeedbackQuestion(questionId));
+            FeedbackQuestionAttributes fqa = fqLogic.getFeedbackQuestion(questionId);
+            if (fqa == null) {
+                allQuestions = new ArrayList<>();
```

Comment:
Might as well standardise to use `Collections` since I'm assuming it's meant to be immutable
```suggestion
                allQuestions = Collections.emptyList();
```